ci(github): [release] run on branch 'master' and 'next'

SimonGolms opened the pull request:
- update to `ubuntu-22.04` runner-image

ElfenB reviewed (commented):
Please check if we can use the new checkout actions (I think it should be v4). The v3 are using node 16 I think and are deprecated.

SimonGolms committed:
ci(github): [release] run on branch 'master' and 'next'

- update to `ubuntu-22.04` runner-image
- fetch all history for all branches and tags
- replace the install dependencies command with the fully written form
- bump action actions/checkout from 3 to 4
- bump action actions/setup-node from 3 to 4
